ldms/src/ldmsd/ldmsd_updtr.c

```diff
@@ -537,28 +548,35 @@ void __ldmsd_prdset_lookup_cb(ldms_t xprt, enum ldms_lookup_status status,
 		assert(NULL == set);
 		status = (status < 0 ? -status : status);
 		if (status == ENOMEM) {
-			ldmsd_log(LDMSD_LERROR,
-				"Error %d in lookup callback for set '%s' "
-				"Consider changing the -m parameter on the "
-				"command line to a larger value. "
-				"The current value is %s\n",
-				status, prd_set->inst_name,
-				ldmsd_get_max_mem_sz_str());
+			if (! (prd_set->warned & LDMSD_PRDCR_SET_WARNED_NOMEM)) {
+				prd_set->warned |= LDMSD_PRDCR_SET_WARNED_NOMEM;
+				ldmsd_log(LDMSD_LERROR,
+					"Error %d in lookup callback for set '%s' "
+					"Consider changing the -m parameter on the "
+					"command line to a larger value. "
+					"The current value is %s\n",
+					status, prd_set->inst_name,
+					ldmsd_get_max_mem_sz_str());
+			}
 		} else if (status == EEXIST) {
-			ldmsd_log(LDMSD_LERROR, "Prdcr '%s': lookup failed asynchronously. "
-					"The set '%s' already exists. "
-					"It is likely that there are more "
-					"than one producers pointing to "
-					"the set.\n",
-					prd_set->prdcr->obj.name, prd_set->inst_name);
+			warn_once_prdcr_set_exists(prd_set);
 		} else {
-			ldmsd_log(LDMSD_LERROR,
-				  "Error %d in lookup callback for set '%s'\n",
-					  status, prd_set->inst_name);
+			if (! (prd_set->warned & LDMSD_PRDCR_SET_WARNED_LOOKUP)) {
+				prd_set->warned |= LDMSD_PRDCR_SET_WARNED_LOOKUP;
+				ldmsd_log(LDMSD_LERROR,
+					  "Error %d in lookup callback for set '%s'\n",
+						  status, prd_set->inst_name);
+			}
 		}
 		prd_set->state = LDMSD_PRDCR_SET_STATE_START;
 		goto out;
 	}
+	if (prd_set->warned != 0) {
```

Comment:
No need to tell anyone we're clearing the error, just set it to zero.

Reply:
Again, clear messages were found admin-friendly in v3.